types/relay-compiler/lib/core/GraphQLIRVisitor.d.ts

```diff
@@ -109,4 +109,29 @@ export type NodeVisitor =
           Variable?: NodeVisitorObject<Variable>;
       };
 
-export function visit(root: VisitNode, visitor: NodeVisitor): any;
+export type EnumTypeID = any;
+export type FieldID = any;
+export type ScalarTypeID = any;
+export type TypeID = any;
+
+export interface Schema {
+    expectStringType: () => ScalarTypeID;
+    getEnumValues: (type: EnumTypeID) => string[];
+    getFieldName: (field: FieldID) => string;
+    getFields: (type: TypeID) => FieldID[];
+    getFieldType: (field: FieldID) => TypeID;
+    getListItemType: (type: TypeID) => TypeID;
+    getNullableType: (type: TypeID) => TypeID;
+    getTypeString: (type: TypeID) => string;
+    isAbstractType: (type: TypeID) => boolean;
+    isEnum: (type: TypeID) => boolean;
+    isInputObject: (type: TypeID) => boolean;
+    isInterface: (type: TypeID) => boolean;
+    isList: (type: TypeID) => boolean;
+    isNonNull: (type: TypeID) => boolean;
+    isObject: (type: TypeID) => boolean;
+    isScalar: (type: TypeID) => boolean;
+    isUnion: (type: TypeID) => boolean;
+}
+
+export function visit(schema: Schema, root: VisitNode, visitor: NodeVisitor): any;
```

Comment:
Are u sure about this? https://github.com/facebook/relay/blob/eb2dea7d9d610c432462f900c24ebcbb56de6a97/packages/relay-compiler/core/GraphQLIRVisitor.js#L154-L158